src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java

```diff
@@ -118,6 +118,8 @@ public ReplicateSupplyService(ReplicatesService replicatesService,
             contributionTimeoutTaskDetector.detect();
```

Comment:
This method could take a lot of time, increasing chances to have an task status from the beginning until the end. It should be async in order to delegate responsibility outside.

Reply:
Are you sure ? This method will:
* Fetch all tasks `INITIALIZED` or `RUNNING` from the database
* Call `TaskUpdateManager#publishUpdateTaskRequest` which call `TaskUpdateRequestManager#publishRequest`
The latter is already async.
But it can be made async to avoid impact from refactoring on `TaskUpdateManager` or `TaskUpdateRequestManager`.

Reply:
Don't you agree `contributionTimeoutTaskDetector.detect()` could take quite some time before returning?

Reply:
The `fetch` task could take some time, sure. By the way, if another worker asks for a new replicate to compute at the same time, it'll trigger another `contributionTimeoutTaskDetector.detect()` call. We should add some mechanism to avoid that scenario.
(Some way to prevent `contributionTimeoutTaskDetector.detect()` from starting twice in parallel?)

Reply:
For me, there is a problem with the current implementation which mixes:
* Scheduled async executions in `ContributionTimeoutTaskDetector`
* Direct calls to `ContributionTimeoutTaskDetector#detect`
I think the whole thing should be refactored and I would like to create and fire `ContributionTimeoutTaskEvent` instances and handle those events with a listener in `ContributionTimeoutTaskDetector`.
See https://www.baeldung.com/spring-events

Reply:
Yes, instead of triggering the whole detector (which re-read tasks) we should publish the existing `com.iexec.core.task.event.ContributionTimeoutEvent` when we found a `task.isContributionDeadlineReached()` from the filter since we are sure we got one.

Reply:
I would like to address this in a next PR if it's okay for everyone.